src/DropdownFilter.webmodeler.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as classNames from "classnames";
+import { Alert } from "./components/Alert";
+
+import { DropdownFilter, DropdownFilterProps } from "./components/DropdownFilter";
+import { ContainerProps, ContainerState } from "./components/DropdownFilterContainer";
+import { Utils, parseStyle } from "./utils/ContainerUtils";
+
+// tslint:disable-next-line class-name
+export class preview extends Component<ContainerProps, ContainerState> {
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listviewAvailable: true };
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-dropdown-filter", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            this.renderAlert(),
+            this.renderDropdownFilter()
+        );
+    }
+
+    componentDidMount() {
+        this.validateConfigs();
+    }
+
+    componentWillReceiveProps(_newProps: ContainerProps) {
+        this.validateConfigs();
+    }
+
+    private validateConfigs() {
+        // validate filter values if filterby is attribute, then value should not be empty or "" or " ".
+        const routeNode = findDOMNode(this) as HTMLElement;
+        const targetNode = Utils.findTargetNode(routeNode);
+
+        if (targetNode) {
+            this.setState({ targetNode });
+        }
+        this.setState({ listviewAvailable: true });
+    }
+
+    private renderAlert() {
+        const errorMessage = Utils.validate({
+            ...this.props as ContainerProps,
+            filterNode: this.state.targetNode,
+            filters: this.props.filters,
+            targetListView: this.state.targetListView,
+            validate: !this.state.listviewAvailable
+        });
+
+        return createElement(Alert, {
+            bootstrapStyle: "danger",
+            className: "widget-dropdown-filter-alert",
+            message: errorMessage ? errorMessage : null
```

Comment:
error message returns "" show it will not need to add a ternary operator

Reply:
Done

Reply:
Done